Sources/_TestingInternals/Discovery.cpp

```diff
@@ -301,13 +303,96 @@ static void enumerateTypeMetadataSections(const SectionEnumerator& body) {
     unsigned long size = 0;
     const void *section = getsectiondata(mh, SEG_TEXT, "__swift5_types", &size);
     if (section && size > 0) {
-      body(section, size);
+      bool stop = false;
+      body(mh, section, size, &stop);
+      if (stop) {
+        break;
+      }
+    }
+  }
+}
+
+#elif defined(_WIN32)
+#pragma mark - Windows implementation
+
+/// Find the section with the given name in the given module.
+///
+/// - Parameters:
+///   - module: The module to inspect.
+///   - sectionName: The name of the section to look for. Long section names are
+///     not supported.
+///
+/// - Returns: A pointer to the start of the given section along with its size
+///   in bytes, or `std::nullopt` if the section could not be found. If the
+///   section was emitted by the Swift toolchain, be aware it will have leading
+///   and trailing bytes (`sizeof(uintptr_t)` each.)
+static std::optional<std::pair<const void *, size_t>> findSection(HMODULE module, const char *sectionName) {
```

Comment:
This is technically the segment as per macho. What macho considers sections are lost.

Reply:
The API and Microsoft's documentation refers to them as [sections](https://learn.microsoft.com/en-us/windows/win32/debug/pe-format#section-table-section-headers) either way.

Reply:
Yes, ELF and PE/COFF do not have segment/section, they only have sections.